tests/aggregation_tests.py

```diff
@@ -316,9 +364,139 @@ def test_aplified_count_when_grouped(self):
+        '''
+        Ckecks if ref data is loaded in tem tables
+        '''
+        self.cursor.execute('SELECT * FROM data__risk___risk')
+        self.assertEqual(self.cursor.fetchone(),
+                         (0.0, u'test-risk', u'Test Risk', 0.13456, u'Nice\nSmall\nDescription'))
+        self.cursor.execute('SELECT * FROM data__country___country')
+        self.assertEqual(self.cursor.fetchone(),
+                         (u'AA', u'Test country', u'test-country', u'test-regiton', u'test-continent'))
+        self.cursor.execute('SELECT * FROM data__asn___asn')
+        self.assertEqual(self.cursor.fetchone(), (111111.0, u'Test title', u'AA'))
+
+
+    def test_rds_tables_created(self):
+        '''
+        Cheks if all rds tables are created and temptables renamed
+        '''
+        main.create_rds_tables()
+        for table in self.tablenames:
+            self.cursor.execute(
+                'select exists(select * from information_schema.tables where table_name=%(table)s)',
+                {'table': table})
+            self.assertEqual(self.cursor.fetchone()[0], True)
+
+
+    def test_populate_rds_tables(self):
+        '''
+        Cheks if rds tables are populated
+        '''
+        message = 'Table {table} is empty'
+        # Create tables and pouplate fact_cunt
+        main.create_rds_tables()
+        self.cursor.copy_expert("COPY fact_count from STDIN csv header", StringIO(self.counts))
+        # After running populate tables
+        main.populate_tables()
+        # Non of the tables should be empty (should not return None)
+        for table in self.tablenames:
+            self.cursor.execute('SELECT * FROM %(table)s LIMIT 1',{"table": AsIs(table)})
+            self.assertNotEqual(self.cursor.fetchone(), None, msg=message.format(table=table))
+
+
+    def test_create_constraints(self):
+        '''
+        Cheks if all constraints are created
+        '''
+        c_names = [
+            'dim_risk_pkey', 'dim_country_pkey', 'dim_asn_pkey', 'dim_time_pkey',
+            'fk_country_asn', 'fk_count_risk', 'fk_count_country', 'fk_count_asn',
+            'fk_count_time', 'fk_cube_risk_week','fk_cube_risk_month', 'fk_cube_risk_quarter',
+            'fk_cube_risk_year', 'fk_cube_country_week', 'fk_cube_country_month',
+            'fk_cube_country_quarter','fk_cube_country_year']
+        message = 'Constraint {c_name} is not created'
+        # Create tables and pouplate them
+        main.create_rds_tables()
+        self.cursor.copy_expert("COPY fact_count from STDIN csv header", StringIO(self.counts))
+        main.populate_tables()
+        main.create_constraints()
+        for c_name in c_names:
+            self.cursor.execute("select * from information_schema.table_constraints WHERE constraint_name = %(c_name)s",
+                                {'c_name': c_name})
+            self.assertTrue(self.cursor.fetchone(), msg=message.format(c_name=c_name))
```

Comment:
The general gotcha is to avoid assertions inside for-loops. They are harder to debug when fail. Better select all you need in one query and assert that it equals expected results.